### PR TITLE
TDEVOPS-1968 | add terraformApply for convoy-dev-deployment

### PR DESCRIPTION
**Pull request by varenyatessell**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

Dependent Services #
- [ ] Governance
- [ ] IAM
- [ ] Deployer
- [ ] DB Systems
- [ ] Tenant
- [ ] Software Library
- [ ] Backup
- [ ] Conductor
- [ ] Monitoring
- [ ] DMM

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

**Environment**:

- [ ] Dev
- [ ] Staging

# Checklist:

- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: .github/ci-scripts/bash/functions-for-build [hunk 1]
```diff
@@ -91,6 +117,43 @@ setup_qa_env() {
...
+  EXTENSION="$2"
+  aws_configure_devqa_infra
+  ARTIFACT_FILE=$PWD/build/${ARTIFACT}.${EXTENSION}
+  mkdir -p build
+  cd terraform
+  create_tf_backend "tf-state-convoy-dev-deployment" "tf-state-convoy-dev-deployment/terraform.tfstate-md5" "${COMMON_INFRA_ACCESS_KEY}" "${COMMON_INFRA_SECRET_KEY}"
+  terraform init
+  terraform plan
+  terraform apply -auto-approve
+  terraform_build $PWD ${ARTIFACT_FILE}
```
Comment (sundeeptessell) [inline .github/ci-scripts/bash/functions-for-build:151]: Why do we need to create zip and push?

File: .github/ci-scripts/bash/functions-for-build-camel-case [hunk 2]
```diff
@@ -1,5 +1,13 @@
 #!/bin/bash
 
+awsConfigureDevqaInfra() {
+  aws_configure_devqa_infra "$@"
+}
+
+createTFBackend() {
```
Comment (sundeeptessell) [inline .github/ci-scripts/bash/functions-for-build-camel-case:7]: Do you need this?

File: .github/workflows/master_build_merge.yml [hunk 3]
```diff
@@ -1060,7 +1060,11 @@ jobs:
       ARTIFACTS_DEV_SECRET_KEY: ${{ secrets.ARTIFACTS_DEV_SECRET_KEY }}
       ARTIFACTS_DEV_S3: ${{vars.ARTIFACTS_DEV_S3}}
       BASE_BRANCH: "${{github.event.pull_request.base.ref}}"
-      CI-BRANCH: main
+      CI-BRANCH: TDEVOPS-1968
```
Comment (sundeeptessell) [inline .github/workflows/master_build_merge.yml:1063]: Undo before merge

File: .github/workflows/master_create_tag.yml [hunk 4]
```diff
@@ -190,7 +190,7 @@ jobs:
               validate_label=true
               if [[ "${{ github.event.pull_request.base.ref }}" == "main" ]]; then
                 APP_GROUP=$(yq '.appGroup // "tessell"' convoy.yaml)
-                URL="http://${{vars.CONVOY_API_ENDPOINT}}/devops/applications/app-groups/$APP_GROUP/latest-main-release-label"
+                URL="http://${{vars.CONVOY_DEV_API_ENDPOINT}}/devops/applications/app-groups/$APP_GROUP/latest-main-release-label"
```
Comment (sundeeptessell) [inline .github/workflows/master_create_tag.yml:193]: undo before merge.

File: .github/workflows/master_merge.yml [hunk 5]
```diff
@@ -27,7 +27,7 @@ jobs:
           if [ -f convoy.yaml ]; then
             if [[ "${{ github.event.pull_request.base.ref }}" == "main" ]]; then
               APP_GROUP=$(yq '.appGroup // "tessell"' convoy.yaml)
-              URL="http://${{vars.CONVOY_API_ENDPOINT}}/devops/applications/app-groups/$APP_GROUP/latest-main-release-label"
+              URL="http://${{vars.CONVOY_DEV_API_ENDPOINT}}/devops/applications/app-groups/$APP_GROUP/latest-main-release-label"
```
Comment (sundeeptessell) [inline .github/workflows/master_merge.yml:30]: undo before merge

File: .github/workflows/master_merge.yml [hunk 6]
```diff
@@ -54,15 +54,15 @@ jobs:
   
   create-tag:
     needs: runner-check
-    uses: TessellDevelopment/convoy-ci/.github/workflows/master_create_tag.yml@main
+    uses: TessellDevelopment/convoy-ci/.github/workflows/master_create_tag.yml@TDEVOPS-1968
```
Comment (sundeeptessell) [inline .github/workflows/master_merge.yml:57]: undo before merge

File: .github/ci-scripts/bash/functions-for-build (hunk 1 above)
Comment (varenyatessell) [reply .github/ci-scripts/bash/functions-for-build:151]: No particular need but just to keep a track, can be removed.

File: .github/ci-scripts/bash/functions-for-build-camel-case (hunk 2 above)
Comment (varenyatessell) [reply .github/ci-scripts/bash/functions-for-build-camel-case:7]: Yes

**Review by sundeeptessell — DISMISSED** [review]

**Review by sundeeptessell — APPROVED** [review]

**Pull request merged**